iceoryx_examples/automotive_soa/include/owl/kom/field_subscriber.inl

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_EXAMPLES_AUTOMOTIVE_SOA_FIELD_SUBSCRIBER_INL
+#define IOX_EXAMPLES_AUTOMOTIVE_SOA_FIELD_SUBSCRIBER_INL
+
+#include "owl/kom/field_subscriber.hpp"
+
+namespace owl
+{
+namespace kom
+{
+template <typename T>
+inline FieldSubscriber<T>::FieldSubscriber(const ServiceIdentifier& service,
+                                           const InstanceIdentifier& instance,
+                                           const FieldIdentifier& field) noexcept
+    : m_subscriber({service, instance, field}, {QUEUE_CAPACITY, HISTORY_REQUEST})
+    , m_client({service, instance, field})
+{
+    m_waitset.attachState(m_client, iox::popo::ClientState::HAS_RESPONSE).expect("Failed to attach client!");
+}
+
+template <typename T>
+inline FieldSubscriber<T>::~FieldSubscriber() noexcept
+{
+    m_waitset.detachState(m_client, iox::popo::ClientState::HAS_RESPONSE);
+
+    // Wait here if a thread or callback is still running
+    while (m_threadsRunning > 0)
+    {
+    }
```

Comment:
Inside the loop I would add `std::thread::yield()` to inform other threads that they can run and to make this loop a little less busy.

Reply:
Great idea, added.